README.md

```diff
@@ -63,16 +63,35 @@ After making your code changes, run the tests and CI checks:
 make ci
 ```
 
-If you want to test your changes, create a new image with the version set to `dev`:
+### Run Locally
+
+If you want to run `digitalocean-cloud-controller-manager` locally against a
+particular cluster, keep your kubeconfig ready and start the binary in the main
+package-hosted directory like this:
+
+```bash
+cd cloud-controller-manager/cmd/digitalocean-cloud-controller-manager
+FAKE_REGION=fra1 go run main.go --kubeconfig <path to your kubeconfig file> --leader-elect=false --v=5 --cloud-provider=digitalocean
+```
+
+The `FAKE_REGION` environment variable takes a (valid) DigitalOcean region. It
+is needed to keep `digitalocean-cloud-controller-manager` from trying to access
+the DigitalOcean metadata service which is only available on droplets. Overall,
+which region you choose should not matter a lot as long as you pick one.
+
+### Run Containerized
+
+If you want to test your changes in a containerized environment, create a new
+image with the version set to `dev`:
 
 ```bash
 VERSION=dev make publish
 ```
 
 This will create a binary with version `dev` and docker image pushed to
-`digitalocean/digitalocean-cloud-controller-manager:dev`
+`digitalocean/digitalocean-cloud-controller-manager:dev`.
```

Comment:
Does it assume that a person who runs `make publish` has access to `digitalocean/digitalocean-cloud-controller-manager` images repository on Docker Hub? Would it make sense to retrieve Git SHA and make `VERSION=` env optional?

Reply:
I also see that `make build` exists. I wonder if mentioning it in README along with / instead of `make publish` would be useful for external contributors.

Reply:
It is possible to push to a different Docker Hub organization by setting the `REGISTRY` environment variable (I know, the name is misleading).

I wouldn't mind a PR to document this properly. Bonus points for mimicking what we have in the csi-digitalocean repository where [a single `DOCKER_REPO` variable can be used to override both the organization and the repository part](https://github.com/digitalocean/csi-digitalocean/blob/690eca5abea5724b87a75311fe78a85cc62765b7/Makefile#L31).

Reply:
I'm not super sure about using a git SHA by default. Personally, I think I have never specified one -- I prefer using a readable tag that clearly identifies my feature and overwrite the tag as needed. (All in my personal Docker Hub repo only of course where it's okay to do such things.) But I understand that YMMV, so if there's more demand for it I'm open to letting myself get convinced.

Reply:
PRs to improve our docs on how to use the Makefile targets are more than welcome!

Reply:
Thanks for clarification, makes sense :+1: